lib/bot.js

```diff
@@ -188,22 +188,25 @@ class Bot {
         logger.info(`Next announcement in: ${moment.duration(nextInterval).humanize()}`);
         this.nextAnnounce = setTimeout(() => {
           logger.info('Announcing on schedule to:', this.config.app.announce.channels);
+          // find the channel id
+          let channelIds = {};
+          this.lookup.forEach((item) => {
+            // only announce on channel or group
+            if (
+              (item.is_channel || item.is_group) && // channel or group
+              self.config.app.announce.channels.indexOf(item.name) !== -1
+            ) {
+              channelIds[item.name] = item.id;
+            }
+          });
           // loop through channels and announce
-          self.config.app.announce.channels.forEach((channel) => {
-            // find the channel id
-            this.lookup.forEach((item) => {
-              // only announce on channel or group
-              if (
-                (item.is_channel || item.is_group) && // channel or group
-                self.config.app.announce.channels.indexOf(item.name) !== -1
```

Comment:
Instead of all these changes, couldn't you just change this line to be: `channel === item.name` ?